modules/gke-ap/main.tf

```diff
@@ -0,0 +1,179 @@
+  project = var.project
+
+  network    = var.network
+  subnetwork = var.subnetwork
+
+  location       = var.region
+  node_locations = var.zones
+
+  deletion_protection = var.deletion_protection
+
+  enable_autopilot = true
+
+  # Use Dataplane V2 (eBPF based networking)
+  datapath_provider = "ADVANCED_DATAPATH"
+
+  networking_mode = "VPC_NATIVE"
+  // Keeping this empty means GKE handles the secondary pod/service CIDR creation
+  ip_allocation_policy {}
+
+  workload_identity_config {
+    workload_pool = "${var.project}.svc.id.goog"
+  }
+
+  release_channel {
+    # NOTE: Toggle to "RAPID" when we want to start playing with things like gcsfuse
+    channel = var.release_channel
+  }
+
+  # Configured with separate node_pool resources
+  # node_config {}
+
+  # https://github.com/hashicorp/terraform-provider-google/issues/9505#issuecomment-1340074019
+  cluster_autoscaling {
+    auto_provisioning_defaults {
+      service_account = google_service_account.cluster_default.email
+    }
+  }
+
+  master_authorized_networks_config {
+    # gcp_public_cidrs_access_enabled = true
+    cidr_blocks {
+      display_name = "Everywhere"
+      cidr_block   = "0.0.0.0/0"
+    }
+
+    # TODO: Pin this to https://api.github.com/meta
+    # Github recommends against doing this, so maybe there's a more effective way, perhaps a certain scale with a tail?
+    # cidr_blocks {}
+  }
+
+  private_cluster_config {
+    enable_private_nodes = var.enable_private_nodes
+    master_global_access_config {
+      enabled = true
+    }
+    # This doesn't do what you think it does
+    # private_endpoint_subnetwork = var.subnetwork
+  }
+
+  dns_config {
+    # Enable more efficient DNS resolution by leveraging the GCP backplane (instead of kube-dns)
+    # Technically this adds cloud DNS billing, but the cost is negligible
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/cloud-dns
+    cluster_dns       = "CLOUD_DNS"
+    cluster_dns_scope = "CLUSTER_SCOPE"
+  }
+
+  # TODO: These probably could be configurable
+  addons_config {
+    http_load_balancing {
+      disabled = false
+    }
+    gke_backup_agent_config {
+      enabled = false
+    }
+    config_connector_config {
+      enabled = false
+    }
+    gcs_fuse_csi_driver_config {
+      enabled = true
+    }
+  }
+
+  monitoring_config {
+    enable_components = ["SYSTEM_COMPONENTS", "APISERVER", "SCHEDULER", "CONTROLLER_MANAGER", "STORAGE", "POD"]
+    managed_prometheus { enabled = true }
+
+  }
+
+  # This can't hurt... right?
+  cost_management_config {
+    enabled = true
+  }
+
+  timeouts {
+    create = "30m"
+    update = "30m"
+    delete = "30m"
+  }
+
+  lifecycle {
+    ignore_changes = [initial_node_count]
+  }
+
+  depends_on = [google_service_account.cluster_default]
+}
+
+# Allow GKE master to hit non 443 ports for webhook/admission controllers
+#
+# https://github.com/kubernetes/kubernetes/issues/79739
+resource "google_compute_firewall" "master_webhook" {
```

Comment:
is this only necessary if we use webhook/admission controller?
if so, make it configurable?

Reply:
I just ripped this from the existing gke module. its bitten me enough times where I'm of the mind of just to default allow, but I'll cave if you feel strongly about it and expose a knob for it 🙏

Reply:
im just worried about carrying cruft for legacy reasons

do we really want to support webhook/admission controllers on autopilot instances? how does the scaling work with webhook/admission controller

i guess my take is, lets not include stuff for just in case
seems like bad engineering practice to include stuff for things you might need but never use

Reply:
would have made this a conditional resource with a variable to enable/disable it
we can always add it back later